lua/bufferline/options.lua

```diff
@@ -42,6 +53,16 @@ function options.closable()
   return get('closable', true)
 end
 
+--- @return bufferline.options.diagnostics
+function options.diagnostics()
+  return vim.tbl_deep_extend('keep', get('diagnostics', {}), {
+    [ERROR] = {enabled = false, icon = 'Ⓧ '},
+    [HINT] = {enabled = false, icon = '💡'},
+    [INFO] = {enabled = false, icon = 'ⓘ '},
+    [WARN] = {enabled = false, icon = '⚠️ '},
+  })
+end
+
```

Comment:
Can we extract the default options as a constant instead? IIUC, we're allocating a new default table for each call to this, even when it's not used. This of course assumes the returned table is not mutated by the rest of the code.

Reply:
I've amended the commit to extract this table out :+1: